lib/rubocop/cop/rspec/sort_metadata.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Sort RSpec metadata alphabetically.
+      #
+      # @example
+      #   # bad
+      #   describe 'Something', :b, :a
+      #   context 'Something', foo: 'bar', baz: true
+      #   it 'works', :b, :a, foo: 'bar', baz: true
+      #
+      #   # good
+      #   describe 'Something', :a, :b
+      #   context 'Something', baz: true, foo: 'bar'
+      #   it 'works', :a, :b, baz: true, foo: 'bar'
+      #
+      class SortMetadata < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Sort metadata alphabetically.'
+
+        # @!method rspec_metadata(node)
+        def_node_matcher :rspec_metadata, <<~PATTERN
+          (block
+            (send
+              #rspec? {#Examples.all #ExampleGroups.all #SharedGroups.all #Hooks.all #Includes.all}
+              _ ${send str sym}* (hash $...)?)
+            ...)
+        PATTERN
+
+        # @!method rspec_configure(node)
+        def_node_matcher :rspec_configure, <<~PATTERN
+          (block (send #rspec? :configure) (args (arg $_)) ...)
+        PATTERN
+
+        # @!method metadata_in_block(node)
+        def_node_search :metadata_in_block, <<~PATTERN
+          (send (lvar $_) #Hooks.all _ ${send str sym}* (hash $...)?)
+        PATTERN
+
+        def on_block(node)
+          if (block_var = rspec_configure(node))
+            metadata_in_block(node).each do |receiver, symbols, pairs|
+              investigate(symbols, pairs.flatten) if receiver == block_var
+            end
+          elsif (symbols, pairs = rspec_metadata(node))
+            investigate(symbols, pairs.flatten)
+          end
```

Comment:
One nice thing about `def_node_matcher` is that It yields the match (if there is one; and the node if it matches, but there are no explicit captures), or doesn't yield if there is no match. So that could be rewritten as follows

```suggestion
          rspec_configure(node) do |block_var|
            metadata_in_block(node).each do |receiver, symbols, pairs|
              investigate(symbols, pairs.flatten) if receiver == block_var
            end
          end
          
          rspec_metadata(node) do |symbols, pairs|
            investigate(symbols, pairs.flatten)
          end
```
Furthermore, the `def_node_matcher` may receive additional arguments (with the use of `%`) so you can add a parameter to `metadata_in_block`:
```
         def_node_search :metadata_in_block, <<~PATTERN
           (send (lvar %) #Hooks.all _ ${send str sym}* (hash $...)?)
         PATTERN
```
And then the code will become:
```
          rspec_configure(node) do |block_var|
            metadata_in_block(node, block_var) do |symbols, pairs|
              investigate(symbols, pairs.flatten)
            end
          end
          
          rspec_metadata(node) do |symbols, pairs|
            investigate(symbols, pairs.flatten)
          end
```